<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="tsv_filename" 								default="ground_truth.tsv" />
	
	<arg name="tsv_time_offset" 							default="0.0" />
	<arg name="tsv_data_columns" 							default="2+18+34" />
	<arg name="tsv_data_multiplier" 						default="0.001" />
	<arg name="tsv_data_offset_x" 							default="0.0" />
	<arg name="tsv_data_offset_y" 							default="0.0" />
	<arg name="tsv_data_offset_z" 							default="0.0" />
	<arg name="tsv_time_multiplier" 						default="1.0" />
	<arg name="tsv_null_string" 							default="NULL" />
	<arg name="load_base_time_from_tsv_header" 				default="false" />

	<arg name="source_frame_ids" 							default="base_link+camera2+camera1" />
	<arg name="target_frame_ids" 							default="map_qualisys+map_qualisys+map_qualisys" />
	<arg name="invert_tf_transforms" 						default="1+0+0" />
	<arg name="invert_tf_hierarchies" 						default="1+0+0" />
	<arg name="pose_topics" 								default="pose_robot+pose_camera2+pose_camera1" />		<!-- use none to ignore one of the tracked bodies -->
	<arg name="pose_array_topics" 							default="poses_robot+poses_camera2+poses_camera1" />	<!-- use none to ignore one of the tracked bodies -->
	<arg name="tsv_data_offset_qx" 							default="0.0" />
	<arg name="tsv_data_offset_qy" 							default="0.0" />
	<arg name="tsv_data_offset_qz" 							default="0.0" />
	<arg name="tsv_data_offset_qw" 							default="1.0" />
	<arg name="number_msgs_to_skip_in_pose_array_msgs" 		default="99+99+99" />


	<node pkg="qualisys_tsv_parsers" type="tsv_to_tf_node" name="tsv_to_tf_node" output="screen" > <!-- launch-prefix="gdbserver localhost:1337" -->
		<param name="tsv_filename" 								type="str" 		value="$(arg tsv_filename)" />
		<param name="tsv_time_offset" 							type="double" 	value="$(arg tsv_time_offset)" />
		<param name="tsv_data_columns" 							type="str" 		value="$(arg tsv_data_columns)" />
		<param name="tsv_data_multiplier" 						type="double" 	value="$(arg tsv_data_multiplier)" />
		<param name="tsv_data_offset_x" 						type="double" 	value="$(arg tsv_data_offset_x)" />
		<param name="tsv_data_offset_y" 						type="double" 	value="$(arg tsv_data_offset_y)" />
		<param name="tsv_data_offset_z" 						type="double" 	value="$(arg tsv_data_offset_z)" />
		<param name="tsv_time_multiplier" 						type="double" 	value="$(arg tsv_time_multiplier)" />
		<param name="tsv_null_string" 							type="str" 		value="$(arg tsv_null_string)" />
		<param name="load_base_time_from_tsv_header" 			type="bool" 	value="$(arg load_base_time_from_tsv_header)" />
		<param name="source_frame_ids" 							type="str" 		value="$(arg source_frame_ids)" />
		<param name="target_frame_ids" 							type="str" 		value="$(arg target_frame_ids)" />
		<param name="invert_tf_transforms" 						type="str" 		value="$(arg invert_tf_transforms)" />
		<param name="invert_tf_hierarchies" 					type="str" 		value="$(arg invert_tf_hierarchies)" />
		<param name="pose_topics" 								type="str" 		value="$(arg pose_topics)" />
		<param name="pose_array_topics" 						type="str" 		value="$(arg pose_array_topics)" />
		<param name="tsv_data_offset_qx" 						type="double" 	value="$(arg tsv_data_offset_qx)" />
		<param name="tsv_data_offset_qy" 						type="double" 	value="$(arg tsv_data_offset_qy)" />
		<param name="tsv_data_offset_qz" 						type="double" 	value="$(arg tsv_data_offset_qz)" />
		<param name="tsv_data_offset_qw" 						type="double" 	value="$(arg tsv_data_offset_qw)" />
		<param name="number_msgs_to_skip_in_pose_array_msgs" 	type="str" 		value="$(arg number_msgs_to_skip_in_pose_array_msgs)" />
	</node>
</launch>
